projects/gateway2/translator/sslutils/ssl_utils.go

```diff
@@ -59,3 +78,50 @@ func cleanedSslKeyPair(certChain, privateKey, rootCa string) (cleanedChain strin
 
 	return cleanedChain, err
 }
+
+// ApplySslExtensionOptions applies the GatewayTLSConfig options to the SslConfig
+// This function will never exit early, even if an error is encountered.
+// It will apply all options and return all errors encountered.
+func ApplySslExtensionOptions(ctx context.Context, in *gwv1.GatewayTLSConfig, out *ssl.SslConfig) error {
+	if len(in.Options) == 0 {
+		return nil
+	}
+
+	var err error
+
+	if oneWayTls, ok := in.Options[GatewaySslOneWayTls]; ok {
+		if strings.ToLower(string(oneWayTls)) == "true" {
+			out.OneWayTls = wrapperspb.Bool(true)
+		}
+	}
+
+	if varifySubjectAltNameStr, ok := in.Options[GatewaySslVerifySubjectAltName]; ok {
+		altNames := strings.Split(string(varifySubjectAltNameStr), ",")
+		out.VerifySubjectAltName = altNames
+	}
+
+	out.Parameters = &ssl.SslParameters{}
+	if cipherSuitesStr, ok := in.Options[GatewaySslCipherSuitesOption]; ok {
```

Comment:
Given that we are pulling from an amorphous map I think we need to consider fatfingering and therefore have some logger or some such?